src/ServerBlock.hpp

```diff
@@ -11,24 +31,45 @@ class ServerBlock
 		ServerBlock& operator=(const ServerBlock& obj);
 		~ServerBlock();
 
-		std::map<std::string, std::string> getVariables() const; // variables outside of locations
-		std::pair<std::string, std::string> getVariables(std::string key) const;
-		std::vector<std::map<std::string, std::string> > getLocations() const; // location / {} blocks
-		
-		void addVariable(std::string& key, std::string& value);
-		void addLocation(std::map<std::string, std::string>& var);
+		bool addVariable(std::string key, std::string& value, bool isLocation);
```

Comment:
Is addVariable something like a setter? Like setVariable?

Reply:
It is used only by Config.hpp, so it is should not be public I guess. Maybe there is a way to give access only to a specific class

Reply:
You can use then maybe the friend keyword. https://en.cppreference.com/w/cpp/language/friend